lib/rucio/core/did_meta_plugins/__init__.py

```diff
@@ -16,186 +16,211 @@
+# True to manages_key() will be used.
+# 
+METADATA_PLUGIN_MODULES = []
+for meta_module_path in METADATA_PLUGIN_MODULE_PATHS:
     try:
         base_module = ".".join(meta_module_path.split(".")[:-1])
         base_class = meta_module_path.split(".")[-1]
-        meta_handler_module = getattr(importlib.import_module(base_module), base_class)()
-        METADATA_HANDLERS.append(meta_handler_module)
+        metadata_plugin_module = getattr(importlib.import_module(base_module), base_class)()
+        METADATA_PLUGIN_MODULES.append(metadata_plugin_module)
     except ImportError:
-        raise PolicyPackageNotFound('Module ' + meta_module_path + ' not found')
+        raise exception.PolicyPackageNotFound('Module ' + meta_module_path + ' not found')
 
 
 def get_metadata(scope, name, plugin="DID_COLUMN", session=None):
     """
     Gets the metadata for given did.
-    This method has been adapted to bring the metadata from diffrent metadata storages. (HARDCODED or GENERIC for now)
-    If the filter is "ALL", will return the metadata from all available metadata storages. Else filter can be used to
-    only return the metadata of a particular storage.
+
+    This method retrieves metadata using different metadata plugins.
+
+    If [plugin] is set to "all", metadata from all available metadata plugins will be returned, 
+    else [plugin] can be used to only return the metadata using a specific plugin.
 
     :param scope: The scope of the did.
-    :param name: The name of the did.
-    :param filter: (optional) Filter down to specific metadata storages [ALL|HARDCODED|GENERIC]
+    :param name: The data identifier name.
+    :param plugin: (optional) Filter specific metadata plugins.
 
     :returns: List of metadata for did.
     """
-    if plugin == "ALL":
-        all_meta = {}
-
-        for meta_handler in METADATA_HANDLERS:
-            metadata = meta_handler.get_metadata(scope, name, session=session)
-            all_meta.update(metadata)
-
-        return all_meta
+    if plugin.lower() == "all":
+        all_metadata= {}
+        for metadata_plugin in METADATA_PLUGIN_MODULES:
+            metadata = metadata_plugin.get_metadata(scope, name, session=session)
+            all_metadata.update(metadata)
+        return all_metadata
     else:
-        for meta_handler in METADATA_HANDLERS:
-            if meta_handler.get_plugin_name().lower() == plugin.lower():
-                return meta_handler.get_metadata(scope, name, session=session)
-
-        raise NotImplementedError('Metadata plugin %s is not enabled on the server.' % plugin)
+        for metadata_plugin in METADATA_PLUGIN_MODULES:
+            if metadata_plugin.get_plugin_name().lower() == plugin.lower():
+                return metadata_plugin.get_metadata(scope, name, session=session)  
+    raise NotImplementedError('Metadata plugin "%s" is not enabled on the server.' % plugin)    #FIXME: this exception isn't returned correctly?
```

Comment:
I don't understand the `FIXME`, is it not correctly handled?

Also, please document that an exception is thrown (`:raises ...: ...`). It was forgotten before.

Reply:
fixme removed. added doc re: raises